lib/rpmsg/rpmsg_internal.h

```diff
@@ -34,6 +34,12 @@ extern "C" {
 	((struct rpmsg_hdr *)((unsigned char *)(p) - sizeof(struct rpmsg_hdr)))
 #define RPMSG_LOCATE_DATA(p) ((unsigned char *)(p) + sizeof(struct rpmsg_hdr))
 
+#define RPMSG_VALIDATE_BUF_ADDR(addr, rvdev, len) \
+	(((addr >= (((struct metal_io_region *)rvdev->rvq->shm_io)->virt + \
+	sizeof(struct rpmsg_hdr))) && \
+	((addr + len) < (((struct metal_io_region *)rvdev->rvq->shm_io)->virt + \
+	((struct metal_io_region *)rvdev->rvq->shm_io)->size))) ? 1 : 0)
+
```

Comment:
The metal_io_region structure should be manipulated in libmetal.
please create `metal_io_block_check ` or `metal_io_is_block_valid` API in libmetal instead

Reply:
@arnopo I have some questions since this will cause a dependency for OpenAMP on a specific version of libmetal.  From a PR merging standpoint, we can just ensure the two PRs in the different repos are merged at the same time, but from a release standpoint, how do we denote that OpenAMP version x is dependent on version y?  Or, do we strive to keep OpenAMP interoperable with older verisons of libmetal?  In which case, I would need to #define the libmetal macro to nothing within OpenAMP if it doesn't already exist in libmetal.

Reply:
Good question:
In term of merging, I will integrate the libmetal first, and rerun the OpenAMP build test to ensure that is working.

For the release. I don't expect to maintain interoperability with older versions of libmetal. 
I think we should only have a dependency message in the release note and offer to send a PR on the dedicated branch for the release if backwards compatibility is needed for a specific libmetal version.
I assume that libmetal APIs will be independent enough to be backported to a legacy branch...
@edmooring: what is your feeling?

Reply:
@arnopo: I agree. I think there has always been an implicit assumption that the libmetal and open amp libraries are tightly coupled. We should probably document this.